internal/model/model.go

```diff
@@ -32,6 +38,88 @@ type Version struct {
 	Model string `json:"model"`
 }
 
+func ParseThingModel(data []byte) (*ThingModel, error) {
+	var tm ThingModel
+	err := json.Unmarshal(data, &tm)
+	if err != nil {
+		return nil, err
+	}
+
+	protos, _ := collectProtocols(data)
+	(&tm).setProtocols(protos)
+	return &tm, nil
+}
+
+func (tm *ThingModel) setProtocols(protos []string) {
+	tm.protocols = protos
+}
```

Comment:
As the function is only called in `ParseThingModel`, I think the protocols on the TM (this one line of code) can be directly set in `ParseThingModel`.